sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/Storage.java

```diff
@@ -281,21 +282,32 @@ public void run() {
 
                 /* If device comes back online. */
                 if (connected) {
-                    for (PendingOperation po : mLocalDocumentStorage.getPendingOperations(Utils.getUserTableName())) {
-                        if (PENDING_OPERATION_CREATE_VALUE.equals(po.getOperation()) ||
-                                PENDING_OPERATION_REPLACE_VALUE.equals(po.getOperation())) {
-                            instanceCreateOrUpdate(po);
-                        } else if (PENDING_OPERATION_DELETE_VALUE.equals(po.getOperation())) {
-                            instanceDelete(po);
-                        } else {
-                            AppCenterLog.debug(LOG_TAG, String.format("Pending operation '%s' is not supported", po.getOperation()));
-                        }
-                    }
+                    processPendingOperations();
                 }
             }
         });
     }
 
+    @Override
+    public void onActivityResumed(Activity activity) {
+        if (mNetworkStateHelper.isNetworkConnected()) {
```

Comment:
Move to applyEnabledState(true) to do it on only start as this tries to sync operations every time you navigate to a page in the app.

Also we need to manage duplicate http calls between that and network listener.